### PR TITLE
bugfix/19808-range-selector-states

### LINKED ISSUE #19808: Range selector week button weird behavior after using the scrollbar.
#### Expected behaviour
When clicking the week button, and scrolling with the scrollbar, the week button should remain active, as the range doesn't change.

#### Actual behaviour
When clicking the week button, and scrolling with the scrollbar, the week button doesn't remain active.

When I change the data range to a smaller one, at some point, the problem doesn't occur. Moreover, I noticed that even when I change the width of the chart, the result might change, see: 

https://github.com/highcharts/highcharts/assets/47400438/f999e071-8e98-4698-a5f2-cdee415d565f

When chart width is 800, everything works correctly, but when I change it to 700, we can see that the button doesn't remain active after scrolling. 

https://jsfiddle.net/BlackLabel/w9y1u8o0/

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/sp5fLr0g/

#### Product version
Highstock v11.1.0

#### Affected browser(s)
All

### PR DESCRIPTION
**Pull request by kamil-musialowski**
Fixed #19808, the range selector week button behaved weirdly after using the scrollbar.

~~Fixed #19808, week/month buttons state was incorrect.~~

- [x] Add test once agreed on a final solution.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205568209738908

### CODE REVIEW

**Comment by kamil-musialowski** [comment]
@pawelfus Pushing the solution draft, let me know what you think

**Review by pawelfus — CHANGES_REQUESTED** [review]
The solution looks fine, but I'm not convinced about the numbers (0.99, 1.01 etc). I think it will be better to keep the strict number of days (5, 28, 365 etc) instead of the percentage of the range.

**Comment by kamil-musialowski** [comment]
@pawelfus Agreed, thank you!

I didn't go with the approach i.e `range > 28 && range < 34` since I wanted this solution to work with buttons that have a `count > 1`, i.e 3M  (months) rangeSelector button.

File: ts/Stock/RangeSelector/RangeSelector.ts [hunk 1]
```diff
@@ -527,7 +527,7 @@ class RangeSelector {
         ): void => {
             const range = rangeOptions._range,
                 type = rangeOptions.type,
-                count = rangeOptions.count || 1,
+                day = 864e5,
```
Comment (pawelfus) [inline ts/Stock/RangeSelector/RangeSelector.ts:530]: Variable already defined in the outer scope

**Review by pawelfus — CHANGES_REQUESTED** [review]
Looks promising!
Missing test + api info

**Comment by kamil-musialowski** [comment]
@pawelfus I've changed the logic a bit - my mistake. Previously I was checking if `range is < > than actualRange +/- few days`, while it should go, `is actualRange < > than a range +/- few days.`

**Review by pawellysy — APPROVED** [review]
Great job! thanks!

**Review by bm64 — APPROVED** [review]
Looks good!

**Review by pawelfus — APPROVED** [review]
Thanks!

**Comment by TorsteinHonsi** [comment]
Are we sure we want the button matching to be sloppier? As noted in the issue, this is a regression after a commit dealing with `fixedRange` on an ordinal axis. Could it be that the solution should be related to the ordinal axis rather than the range selector? Maybe what we need is a more granular fix for 8ec07aca85f67a48c0023da2214f927b961dc4b2 that retains the button state?

**Comment by kamil-musialowski** [comment]
New PR with a new solution can be found here: https://github.com/highcharts/highcharts/pull/20522

**Pull request closed without merging**